src/cargo/ops/cargo_remove.rs

```diff
@@ -59,7 +85,46 @@ pub fn remove(options: &RemoveOptions<'_>) -> CargoResult<()> {
             .warn("aborting remove due to dry run")?;
     } else {
         manifest.write()?;
+
+        if let Some(workspace_manifest) = workspace_manifest {
+            cargo_util::paths::write(
+                workspace.root_manifest(),
+                workspace_manifest.to_string().as_bytes(),
+            )?;
+        }
     }
 
     Ok(())
 }
+
+/// Get whether or not a dependency is marked as `workspace`.
+fn dependency_is_workspace(dep: &str, dep_table: &[String], manifest: &LocalManifest) -> bool {
+    if let Ok(toml_edit::Item::Table(table)) = manifest.get_table(dep_table) {
+        let value = table.get(dep).and_then(|i| i.get("workspace"));
+        if let Some(toml_edit::Item::Value(value)) = value {
+            return value.as_bool() == Some(true);
+        }
+    }
+
+    false
+}
+
+/// Get whether or not a dependency is depended upon in a workspace.
+fn dependency_in_workspace(dep: &str, exclude: &Package, workspace: &Workspace<'_>) -> bool {
+    workspace.members().filter(|p| *p != exclude).any(|p| {
+        p.dependencies()
+            .iter()
+            .any(|d| d.package_name().as_str() == dep)
+    })
+}
```

Comment:
Nope! It would still show up in the package's dependencies here (with a different `DepKind`, but we ignore that), so we would still catch the reference to it.

Reply:
I feel I'm missing something.  It looks like we are checking all members **except** `parent` to see if `child` is present, assuming we are completely removing `child` from `parent` but we aren't as it could be in more than on dependency table.

Reply:
Oh, I see your point. My bad... running on not so much sleep today. :yawning_face: